contracts/near/eth-types/src/eth2.rs

```diff
@@ -63,17 +180,23 @@ pub struct ExtendedBeaconBlockHeader {
     pub execution_block_hash: H256,
 }
 
-impl From<HeaderUpdate> for ExtendedBeaconBlockHeader {
-    fn from(item: HeaderUpdate) -> Self {
-        let root = item.beacon_header.tree_hash_root();
-        ExtendedBeaconBlockHeader {
-            header: item.beacon_header,
-            beacon_block_root: H256(root.0.into()),
-            execution_block_hash: item.execution_block_hash,
+impl From<FinalizedHeader> for ExtendedBeaconBlockHeader {
+    fn from(finalized_header: FinalizedHeader) -> Self {
+        let beacon = finalized_header.beacon;
+        Self {
+            header: beacon.clone(),
+            beacon_block_root: beacon.tree_hash_root().0.into(),
+            execution_block_hash: finalized_header.execution.block_hash,
```

Comment:
What are the advantages of passing the entire block instead of just the hash? It's not aligned with the specification anyway, takes up more memory, and we’re not performing any additional checks.